Python/Product/EnvironmentsList/ToolWindow.xaml.cs

```diff
@@ -363,29 +364,46 @@ await StartRefreshDBAsync(view)
                     select = selectView?.Configuration?.Id;
                 }
 
-                var configs = _interpreters.Configurations.Where(f => f.IsUIVisible());
-                configs = configs.Concat(Enumerable.Repeat(EnvironmentView.OnlineHelpView.Configuration, 1));
-                if (_addNewEnvironmentView != null) {
-                    configs = configs.Concat(Enumerable.Repeat(_addNewEnvironmentView.Configuration, 1));
-                }
+                bool anyMissing = true;
+                for (int retries = 3; retries > 0 && anyMissing; --retries) {
+                    var configs = _interpreters.Configurations.Where(f => f.IsUIVisible());
+                    if (_onlineHelpView != null) {
+                        configs = configs.Concat(Enumerable.Repeat(_onlineHelpView.Configuration, 1));
+                    }
+                    if (_addNewEnvironmentView != null) {
+                        configs = configs.Concat(Enumerable.Repeat(_addNewEnvironmentView.Configuration, 1));
+                    }
 
-                _environments.Merge(
-                    configs,
-                    ev => ev.Configuration,
-                    c => c,
-                    c => {
-                        if (EnvironmentView.IsAddNewEnvironmentView(c.Id)) {
-                            return _addNewEnvironmentView;
-                        } else if (EnvironmentView.IsOnlineHelpView(c.Id)) {
-                            return EnvironmentView.OnlineHelpView;
-                        }
-                        var view = new EnvironmentView(_service, _interpreters, _interpreters.FindInterpreter(c.Id), null);
-                        OnViewCreated(view);
-                        return view;
-                    },
-                    InterpreterConfigurationComparer.Instance,
-                    InterpreterConfigurationComparer.Instance
-                );
+                    anyMissing = false;
+                    _environments.Merge(
+                        configs,
+                        ev => ev.Configuration,
+                        c => c,
+                        c => {
+                            if (EnvironmentView.IsAddNewEnvironmentView(c.Id)) {
+                                return _addNewEnvironmentView;
+                            } else if (EnvironmentView.IsOnlineHelpView(c.Id)) {
+                                return _onlineHelpView;
+                            }
+                            var fact = _interpreters.FindInterpreter(c.Id);
+                            EnvironmentView view = null;
+                            try {
+                                if (fact != null) {
+                                    view = new EnvironmentView(_service, _interpreters, fact, null);
+                                }
+                            } catch (ArgumentException) {
+                            }
```

Comment:
Typically it'll fail when it races with an environment being removed, so we got the ID initially, but by the time we go to create the view it's gone. We'll add a stub item the first time through, and the second time we won't get the ID, so we'll remove the stub item.